### PR TITLE
feat: add participant

### LINKED ISSUE #92: feat: mutation to add participant to an existing trip

### PR DESCRIPTION
**Pull request by AyanavaKarmakar**
https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-26_7c296738-420f-4390-8ab7-1353cf4874d1

### CODE REVIEW

File: src/components/Home/Authenticated/buttons/CopyTripIdButton.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+export const CopyTripIdButton: FC<Props> = (props) => {
+  const { tripId, tripName } = props;
+  const [open, setOpen] = useState(false);
+  const isMd = useMediaQuery("(min-width: 768px)");
+  const { data: session } = useSession();
+
+  const handleCopyToClipboard = async (): Promise<void> => {
+    try {
+      await navigator.clipboard.writeText(
+        `${session?.user?.name} has invited you to join their trip: ${tripName}! The passcode is ${tripId}. Signup / Login at trippins (https://apptypesafetravellers.ayanavakarmakar.repl.co/) to join.`,
```
Comment (AyanavaKarmakar) [inline src/components/Home/Authenticated/buttons/CopyTripIdButton.tsx]:
```suggestion
        `${session?.user?.name} has invited you to join their trip: ${tripName}! The passcode is ${tripId}. Signup / Login at Trippins (https://apptypesafetravellers.ayanavakarmakar.repl.co/) to join.`,
```

**Review by ssarkar551 — APPROVED** [review]

**Comment by Nishith-Savla** [comment]
Great work @AyanavaKarmakar! What I suggest is disabling the confirm button instead of conditional rendering to have consistent UI. Also, I think that renaming `trip passcode` to `trip code` would be better. Thoughts?

**Comment by AyanavaKarmakar** [comment]
@Nishith-Savla done 

https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-26_24d0c629-0ef4-4d9b-8be7-c7897691c166

**Comment by Nishith-Savla** [comment]
> @Nishith-Savla done
> 
> https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-26_24d0c629-0ef4-4d9b-8be7-c7897691c166

Nice! Also disable the hover effect on button with some styles like this:

```
saveEnabled
  ? "border-black bg-green-300 hover:bg-green-400 dark:bg-green-700 dark:hover:bg-green-600 "
  : " border-transparent bg-gray-300 dark:bg-gray-700",
```

**Comment by AyanavaKarmakar** [comment]
https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-26_179ee8cc-2133-47d3-93e0-e79db7b91e58 @Nishith-Savla like this?

**Comment by Nishith-Savla** [comment]
> https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-26_179ee8cc-2133-47d3-93e0-e79db7b91e58 @Nishith-Savla like this?

Yes perfect. I know I'm disturbing you again & again but 1 last improvement: The button moves up and down due to error text which doesn't look good. Can we keep the error text rendered and just change it's content to empty in case of no error?

**Comment by AyanavaKarmakar** [comment]
> > https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-26_179ee8cc-2133-47d3-93e0-e79db7b91e58 @Nishith-Savla like this?
> 
> Yes perfect. I know I'm disturbing you again & again but 1 last improvement: The button moves up and down due to error text which doesn't look good. Can we keep the error text rendered and just change it's content to empty in case of no error?

the text can be made transparent by default and red when it's an error

**Comment by AyanavaKarmakar** [comment]
https://app.flonnect.com/view/video/mccreeper2k/Flonnect_2023-01-27_d51101d6-d993-42e0-a77b-884fc670f888

@Nishith-Savla

File: src/components/Home/Authenticated/buttons/CopyTripIdButton.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+export const CopyTripIdButton: FC<Props> = (props) => {
+  const { tripId, tripName } = props;
+  const [open, setOpen] = useState(false);
+  const isMd = useMediaQuery("(min-width: 768px)");
+  const { data: session } = useSession();
+
+  const handleCopyToClipboard = async (): Promise<void> => {
+    try {
+      await navigator.clipboard.writeText(
+        `${session?.user?.name} has invited you to join their trip: ${tripName}! The trip code is ${tripId}. Signup / Login at Trippins (https://apptypesafetravellers.ayanavakarmakar.repl.co/) to join.`,
```
Comment (AyanavaKarmakar) [inline src/components/Home/Authenticated/buttons/CopyTripIdButton.tsx]:
```suggestion
        `${session?.user?.name} has invited you to join their trip: ${tripName}! The trip code is ${tripId}. Signup / Login at Trippins (https://trippins.ayanavakarmakar.repl.co) to join.`,
```

**Review by Nishith-Savla — APPROVED** [review]

**Pull request merged**